### DISCUSSION TITLE
v1.112.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#11775] [Announcements]
# v1.112.0

## Highlights

Welcome to the release of `v1.112.0` of Immich. This release packs with new UI features and enhancements, and many bugs are getting squashed. We hope you enjoy this release. Some of the key highlights include:

- Material color theming for the mobile app
- Open in map view on the web
- Star rating
- Screen stays on and auto darkens when performing a long session of foreground upload
- Notable fix: Uploading assets from iCloud causing the iOS mobile app to crash
- Notable fix: Photos and videos belonging to a person are now all displayed in the person's view on the mobile app

### Material color theming for the mobile app

The app has had a makeover to prepare for the new school year. You can now choose a different primary color in the app, and the new color will be applied across the app's elements. There are a variety of minor tweaks in UI elements to make the app appear more consistent and aesthetically pleasing.

<p align="center">
<img src="https://github.com/user-attachments/assets/1d3f2547-1aac-412d-907d-a6bed269b64d" width="350" title="material-theme"> 
<img src="https://github.com/user-attachments/assets/8b27d239-4e5f-428a-b61f-493b569c6a6e" width="350" title="material-theme"> 
</p>

### Open in map view on the web

You can now navigate to the map directly from the asset's detail panel by clicking on the map icon at the top right corner of the minimap.

![openinmap](https://github.com/user-attachments/assets/3d672c1b-f0b1-48e9-98dc-5050d83415d7)

### Star rating

As one of the most commonly used pieces of information for photographers, the star rating information in EXIF will now be included in the extracted data. You can run the extract metadata job for all to populate this information in your instance. 

![image](https://github.com/user-attachments/assets/c1ed8eda-1f8e-4764-b50d-658a14c73527)

Enabling this feature in `User Settings > App Settings > Star Rating` lets you view and set the star rating information in the detail panel.

The edited information will be saved in the companion sidecar XMP file.

![settings](https://github.com/user-attachments/assets/447afb3c-e3ba-4121-9d3b-37d78cbfc251)

### Always-on screen for foreground upload

When entering the backup screen and running foreground backup, the screen will now stay on and automatically darken after 30 seconds.

### New search models

Immich now supports a set of newer models with very competitive performance. See #11468 for details and visualizations.

A special mention goes to `ViT-B-16-SigLIP__webli`, which outperforms several much larger ViT-L models despite being the same size as the default model. However, it is a bit slower than the default. At the same size, `ViT-B-16-SigLIP-384__webli` is among the best models period, at the cost of being much slower than other models of its size. It's ideal for memory-constrained environments that still want the best quality.

## Support Immich

<p align="center">
<img src="https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExbjY2eWc5Y2F0ZW56MmR4aWE0dDhzZXlidXRmYWZyajl1bWZidXZpcyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/87CKDqErVfMqY/giphy.gif" width="450" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich by purchasing a product key at [https://buy.immich.app](https://buy.immich.app).

Cheers! 🍻

### 🚀 Features
* feat(server): search unknown place by @jforseth210 in https://github.com/immich-app/immich/pull/10866
* feat(web): open in map view by @jrasm91 in https://github.com/immich-app/immich/pull/11592
* feat(mobile): add support for material themes by @pruthvi-21 in https://github.com/immich-app/immich/pull/11560
* feat(mobile): enable wakelock on backup page by @Saschl in https://github.com/immich-app/immich/pull/11621
* feat(mobile): darken screen on backup page by @Saschl in https://github.com/immich-app/immich/pull/11623
* feat(web): add Exif-Rating by @stumpigit in https://github.com/immich-app/immich/pull/11580
* feat(web): add privacy step in the onboarding by @martabal in https://github.com/immich-app/immich/pull/11359
* feat(web): image editor - panel and cropping by @ilyaChuk in https://github.com/immich-app/immich/pull/11074
### 🌟 Enhancements
* feat(ml): add more search models by @mertalev in https://github.com/immich-app/immich/pull/11468
* feat(web): Add stacking option to deduplication utilities by @i-am-a-teapot in https://github.com/immich-app/immich/pull/11114
* chore(server): support pgvecto.rs 0.3.0 by @mertalev in https://github.com/immich-app/immich/pull/11624
* feat: change web asset detail map to zoom level 12.5 by @zackpollard in https://github.com/immich-app/immich/pull/11643
* feat(web): improve group-tab accessibility by @michelheusschen in https://github.com/immich-app/immich/pull/11647
* feat(web): improve shared link management on mobile by @michelheusschen in https://github.com/immich-app/immich/pull/11720
### 🐛 Bug fixes
* fix: make HTML valid by @benmccann in https://github.com/immich-app/immich/pull/11465
* fix(mobile): Crop presets break crop rectangle #11462  by @Yuvi-raj-P in https://github.com/immich-app/immich/pull/11467
* fix(server): external domain url validation by @martyfuhry in https://github.com/immich-app/immich/pull/11493
* fix(web): websocket over ipv6 by @michelheusschen in https://github.com/immich-app/immich/pull/11508
* fix(web): correctly format future timeline dates by @michelheusschen in https://github.com/immich-app/immich/pull/11506
* fix(web): prevent change-location suggestion race-condition by @oidq in https://github.com/immich-app/immich/pull/11523
* fix(mobile): Naming fix for the edited file by @Yuvi-raj-P in https://github.com/immich-app/immich/pull/11503
* fix(mobile): autofill by @alextran1502 in https://github.com/immich-app/immich/pull/11591
* refactor(server): startup checks for vector extension by @mertalev in https://github.com/immich-app/immich/pull/11559
* fix(web): slideshow on iphone by @michelheusschen in https://github.com/immich-app/immich/pull/11599
* chore(server): remove get person asset limit by @alextran1502 in https://github.com/immich-app/immich/pull/11597
* fix(server): video duration extraction by @michelheusschen in https://github.com/immich-app/immich/pull/11610
* fix(server): handle numeric 'Image Description' and 'Description' values by @jo-gross in https://github.com/immich-app/immich/pull/11636
* fix(mobile): iOS crashing when download iCloud content by @alextran1502 in https://github.com/immich-app/immich/pull/11639
* fix(server): avoid transcoding thumbnail streams by @mincrmatt12 in https://github.com/immich-app/immich/pull/11603
* fix(web): keep album description in sync by @michelheusschen in https://github.com/immich-app/immich/pull/11652
* fix(server): use luxon for maxdate validator by @michelheusschen in https://github.com/immich-app/immich/pull/11651
* chore(mobile): refactor detail panel by @alextran1502 in https://github.com/immich-app/immich/pull/11662
* fix(mobile): incorrect remove action from the album assets detail view by @alextran1502 in https://github.com/immich-app/immich/pull/11671
* fix(web): asset viewer dynamic size by @michelheusschen in https://github.com/immich-app/immich/pull/11697
* fix(web): use fallback image if shared asset isn't resized by @michelheusschen in https://github.com/immich-app/immich/pull/11704
* fix(web): hide import json button when using config file by @michelheusschen in https://github.com/immich-app/immich/pull/11714
* fix(web): detail panel out of sync when reopening by @michelheusschen in https://github.com/immich-app/immich/pull/11713
* fix(web): single row of items by @michelheusschen in https://github.com/immich-app/immich/pull/11729
* fix(server): follow symlinks when zipping assets by @C-Otto in https://github.com/immich-app/immich/pull/11685
* fix(web): fallback aperture info when there is no locale set by @alextran1502 in https://github.com/immich-app/immich/pull/11770
* fix(ml): pydantic dep causes starting up issue by @alextran1502 in https://github.com/immich-app/immich/pull/11773
### 📚 Documentation
* Add Immich Distribution to Community Projects page by @nsg in https://github.com/immich-app/immich/pull/11576
* feat(docs): privacy policy by @alextran1502 in https://github.com/immich-app/immich/pull/11535
* chore: more cursed knowledge by @jrasm91 in https://github.com/immich-app/immich/pull/11630
* chore: more cursed knowledge by @jrasm91 in https://github.com/immich-app/immich/pull/11631
* docs: how to decrease Redis logs by @mmomjian in https://github.com/immich-app/immich/pull/11638
* fix(docs): update documentation by @C-Otto in https://github.com/immich-app/immich/pull/11655
* chore: add warning to media location env var by @bo0tzz in https://github.com/immich-app/immich/pull/11665
* docs: Update docs + example.env for DB_PASSWORD by @mmomjian in https://github.com/immich-app/immich/pull/11678
* docs: update custom map style guide by @pcouy in https://github.com/immich-app/immich/pull/11350
### 🌐 Translations
* feat(web): more languages by @jrasm91 in https://github.com/immich-app/immich/pull/11488
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/11458
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/11533

## New Contributors
* @jforseth210 made their first contribution in https://github.com/immich-app/immich/pull/10866
* @oidq made their first contribution in https://github.com/immich-app/immich/pull/11523
* @nsg made their first contribution in https://github.com/immich-app/immich/pull/11576
* @foxit64 made their first contribution in https://github.com/immich-app/immich/pull/11590
* @pruthvi-21 made their first contribution in https://github.com/immich-app/immich/pull/11560
* @i-am-a-teapot made their first contribution in https://github.com/immich-app/immich/pull/11114
* @jo-gross made their first contribution in https://github.com/immich-app/immich/pull/11636
* @mincrmatt12 made their first contribution in https://github.com/immich-app/immich/pull/11603
* @C-Otto made their first contribution in https://github.com/immich-app/immich/pull/11655
* @stumpigit made their first contribution in https://github.com/immich-app/immich/pull/11580
* @dotlambda made their first contribution in https://github.com/immich-app/immich/pull/11701
* @pcouy made their first contribution in https://github.com/immich-app/immich/pull/11350
* @ilyaChuk made their first contribution in https://github.com/immich-app/immich/pull/11074

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.111.0...v1.112.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.112.0'>v1.112.0</a>.</em>

**Comment by stumpigit** [comment]
Congratulations on the release! I'm already working on expanding the search so that you can filter by star rating... I hope this can be integrated in the next release.

![starratingfilter](https://github.com/user-attachments/assets/efbe4682-35a6-4e8a-8282-93c895474b38)

Many thanks to all contributors and happy Immich!

**Comment by lukanowa** [comment]
Thx for new release:)

chore(server): support pgvecto.rs 0.3.0 by @mertalev in https://github.com/immich-app/immich/pull/11624

Is that means we can change the version in docker-compose?

**Reply by alextran1502** [reply]
If you don't see any issues, I highly suggest you keep the current running version.

**Reply by mertalev** [reply]
Immich will work fine without upgrading and we won't be requiring 0.3.0.

If you'd like to *optionally* change it, the instructions for the default setup are to:
1. Make a backup of the database
2. Change the version in the Compose file to 0.3.0
3. `docker compose up`
4. `docker compose down` after seeing logs about needing to restart Postgres
5. `docker compose up`

For Postgres instances where Immich isn't a superuser:

1. Make a backup of the database
2. Change the version in the Compose file to 0.3.0
3. Start Postgres
4. Run the below commands:
```sql
ALTER EXTENSION vectors UPDATE;
SELECT pgvectors_upgrade();
```
5. Restart Postgres
6. Start Immich

**Comment by SteveDinn** [comment]
Just upgraded to 1.112.0 and I'm seeing this error repeated in my logs. Where do I set this setting? Should it be an environment variable? I'm assuming that it should contain the URL via which the immich instance is accessed, is this correct?
```
[Nest] 17  - 08/14/2024, 6:31:32 PM   ERROR [Api:An instance of SystemConfigDto has failed the validation:
- property server.externalDomain has failed the following constraints: isUrl
~of0o8whp] Validation error
```

**Reply by alextran1502** [reply]
We fixed an issue of possible invalid External Domain settings

![image](https://github.com/user-attachments/assets/102cdebc-3a74-498d-a588-25498191fc26)

Make sure yours has `http://` or `https://`

**Reply by SteveDinn** [reply]
Thanks. That was it. I didn't have the `https://` prefix.

**Comment by iFloris** [comment]
Thanks! Congratulations on another great release. So what is the 'best' new smart search model to use? If I want to use multilingual, would that be `ViT-B-16-SigLIP-i18n-256__webli`? Memory isn't much of a constraint for my setup.

**Reply by mertalev** [reply]
The NLLB models (specifically the new `mrl` variants) are the theoretical best for almost all of them. However, they work best when you start the search query with a specific identifier for the language (e.g. `eng_Latn crashing waves`). I'm unsure how well they work without that.

`XLM-Roberta-Large-ViT-H-14__frozen_laion5b_s13b_b90k` is generally a safe choice as it's the best aside from the NLLB models, supports a wide range of languages and doesn't expect a language token.

**Reply by iFloris** [reply]
Thanks for your advice @mertalev, I've set that up.

**Comment by samluxh** [comment]
Hi there,

thank you for the update, I’m currently using immich and really enjoying it! However, I’ve noticed that it still seems impossible to filter by just a month (for example, August across all years) or by a year without specifying a particular day or date.

Is this functionality still not available, or is there a way to achieve this that I might have missed?

Thanks in advance for your help, and thanks for all the great work on immich!

**Reply by mertalev** [reply]
That's an interesting idea! Feel free to make a feature request for it if there isn't one already.

**Reply by samluxh** [reply]
The feature request #10684 seems to be same/similar, so I just commented there again. Thank you! :-)

**Comment by mio-19** [comment]
For docker compose file provided with release What changed in command? I can't see it
```
     volumes:
+      # Do not edit the next line. If you want to change the database storage location on your system, edit the value
 of DB_DATA_LOCATION in the .env file
       - ${DB_DATA_LOCATION}:/var/lib/postgresql/data
     healthcheck:
       test: pg_isready --dbname='${DB_DATABASE_NAME}' --username='${DB_USERNAME}' || exit 1; Chksum="$$(psql --dbname
='${DB_DATABASE_NAME}' --username='${DB_USERNAME}' --tuples-only --no-align --command='SELECT COALESCE(SUM(checksum_fa
ilures), 0) FROM pg_stat_database')"; echo "checksum failure count is $$Chksum"; [ "$$Chksum" = '0' ] || exit 1
       interval: 5m
       start_interval: 30s
       start_period: 5m
-    command: ["postgres", "-c" ,"shared_preload_libraries=vectors.so", "-c", 'search_path="$$user", public, vectors',
 "-c", "logging_collector=on", "-c", "max_wal_size=2GB", "-c", "shared_buffers=512MB", "-c", "wal_compression=on"]
+    command: ["postgres", "-c", "shared_preload_libraries=vectors.so", "-c", 'search_path="$$user", public, vectors',
 "-c", "logging_collector=on", "-c", "max_wal_size=2GB", "-c", "shared_buffers=512MB", "-c", "wal_compression=on"]
     restart: always

 volumes:
```

**Reply by mertalev** [reply]
I don't recall the command being changed. It's probably just a white space change from formatting.

**Comment by cfelicio** [comment]
Just upgraded the IOS App, and the "local network" option is gone?

**Reply by alextran1502** [reply]
Where can you find that?

**Comment by Ador-able** [comment]
There is no     ghcr.io/immich-app/immich-machine-learning:v1.112.1-openvino  ！！

**Reply by imohitkr** [reply]
yeah hit the same issue.

**Reply by imohitkr** [reply]
@alextran1502 @mertalev can you please help with this? It was working fine earlier with v1.111.0.  My compose file for machine-learning container looks like this: 

```compose
immich-machine-learning:
    container_name: immich_machine_learning
    # user: 999:10
    # security_opt:
    #   - no-new-privileges:true
    labels:
      com.centurylinklabs.watchtower.enable: "true"
      com.centurylinklabs.watchtower.monitor-only: "true"
    # For hardware acceleration, add one of -[armnn, cuda, openvino] to the image tag.
    # Example tag: ${IMMICH_VERSION:-release}-cuda
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}-openvino
    # extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/ml-hardware-acceleration
    #   file: hwaccel.ml.yml
    #   service: cpu # set to one of [armnn, cuda, openvino, openvino-wsl] for accelerated inference - use the `-wsl` version for WSL2 where applicable
    device_cgroup_rules:
      - 'c 189:* rmw'
    devices:
      - /dev/dri:/dev/dri
    volumes:
      - /dev/bus/usb:/dev/bus/usb
      - /volume2/docker/immich/model_cache:/cache
    env_file:
      - stack.env
    restart: always
```

**Reply by alextran1502** [reply]
It looks like there is something wrong with the debian repository so the build cannot finish, you guys can temporarily pin the version to v1.111.0 for the machine learning image tag

**Comment by raisinbear** [comment]
Thanks a lot for all your efforts keeping this going forward! 🎉
Great to see new models, too! 😀 Is the performance penalty mentioned restricted to classification only or will it slow down search as well?

**Reply by mertalev** [reply]
It will slow down search too. Pro tip: if you want to avoid the delay the first time you search and you're okay with keeping the model loaded, add the env `MACHINE_LEARNING_PRELOAD__CLIP=<model-name>`. It will load the model at startup instead of waiting for a request.

**Reply by raisinbear** [reply]
Thanks a lot. Seems like I’ll have to try then. Also great tip with the preloading!

**Comment by nnmalex** [comment]
May I suggest that auto dimming of the screen can be annoying and would be nice as an opt-in option?
At the same time the fact that the app can stay in foreground and do a large sync is amazing.

**Comment by Weuxel** [comment]
Hi,
maybe add a hint about breaking changes about this?
```

Starting microservices worker
[Nest] 7  - 08/15/2024, 7:50:23 AM     LOG [Microservices:EventRepository] Initialized websocket server
Error: 
    The pgvecto.rs extension version is 0.0.0, which means it is a nightly release.

    Please run 'DROP EXTENSION IF EXISTS vectors' and switch to a release version.
    See https://immich.app/docs/guides/database-queries for how to query the database.
    at /usr/src/app/dist/services/database.service.js:83:23
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /usr/src/app/dist/repositories/database.repository.js:186:23
microservices worker exited with code 1
```
Thanks.

**Reply by mertalev** [reply]
That check was added a long time ago. What version are you upgrading from?

**Reply by Weuxel** [reply]
I upgraded from the previous version v1.111.0.

**Reply by mertalev** [reply]
We did refactor this recently. I guess it wasn't working before 😅

**Reply by Weuxel** [reply]
Alright then :wink: 
Thanks for the info. :+1:
